### ISSUE TITLE
missing input: false positive when using an input named "args"

### ISSUE THREAD

**Issue by avx-rvandegrift** [rhysd/actionlint#550]
If a composite action takes an input named `args`, actionlint seems to think it's always unused.  In a fresh, empty repo:

```
$ cat .github/actions/wooo/action.yml
name: wooo
description: wooo

inputs:
  args:
    required: true

runs:
  using: "composite"
  steps:
    - shell: bash
      run: exit 0
$ cat .github/workflows/yeah.yml
name: yeah

on:
  push:

jobs:
  one:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4

      - uses: ./.github/actions/wooo
        with:
          args: heyy
$ actionlint -verbose
verbose: Linting all workflow files in repository: /home/rvandegrift/tmp/repo
verbose: Detected project: /home/rvandegrift/tmp/repo
verbose: Collected 1 YAML files
verbose: Linting .github/workflows/yeah.yml
verbose: Using project at /home/rvandegrift/tmp/repo
verbose: Found 0 parse errors in 0 ms for .github/workflows/yeah.yml
verbose: Rule "pyflakes" was disabled: exec: "pyflakes": executable file not found in $PATH
verbose: Found total 1 errors in 0 ms for .github/workflows/yeah.yml
.github/workflows/yeah.yml:12:15: missing input "args" which is required by action "wooo" defined at "./.github/actions/wooo". all required inputs are "args" [action]
   |
12 |       - uses: ./.github/actions/wooo
   |               ^~~~~~~~~~~~~~~~~~~~~~

```

It passes if I change the name of the input:
```
$ sed -i s/args/myargs/g .github/actions/wooo/action.yml .github/workflows/yeah.yml
$ actionlint -verbose
verbose: Linting all workflow files in repository: /home/rvandegrift/tmp/repo
verbose: Detected project: /home/rvandegrift/tmp/repo
verbose: Collected 1 YAML files
verbose: Linting .github/workflows/yeah.yml
verbose: Using project at /home/rvandegrift/tmp/repo
verbose: Found 0 parse errors in 0 ms for .github/workflows/yeah.yml
verbose: Rule "pyflakes" was disabled: exec: "pyflakes": executable file not found in $PATH
verbose: Found total 0 errors in 0 ms for .github/workflows/yeah.yml
```

**Comment by holtkampjs** [comment]
@avx-rvandegrift It appears that `args` is a reserved key of sorts which is on the `with:` object by default. https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#jobsjob_idstepswithargs

I assume that is the reason for this issue.
